content/ErrorHandling/ErrorHandling.q.md

```diff
@@ -0,0 +1,1292 @@
+
+Correct the code so that it outputs the string `Warning: Divide(): The value '0' as a divisor is not allowed in /index.php on line 5`.
+
+```php
+<?php
+function Divide($dividend, $divisor)
+{
+    if ($divisor == 0) {
+        trigger_error("Divide(): The value '0' as a divisor is not allowed" E_USER_WARNING);
+        return false;
+    } else {
+        return($dividend / $divisor);
+    }
+}
+
+echo Divide(10, 0);
+?>
+```
+
+
+:::
+
+/// type=REPL, readonly=true
+
+```php
+<?php
+function Divide($dividend, $divisor)
+{
+    if ($divisor == 0) {
+        trigger_error( , E_USER_WARNING);
+        return false;
+    } else {
+        return($dividend / $divisor);
+    }
+}
+
+echo Divide(10, 0);
+?>
+```
+/// type=SS, answer=[1]
+
+Execute the program. What is the error message?
+
+ - syntax error, unexpected `','` on line number 5
+
+ - syntax error, unexpected `':'`, expecting `')'` on line number 5
+
+ - Warning: Divide(): The value '0' as a divisor is not allowed in /index.php on line 5
+
+ - syntax error, unexpected `'E_USER_WARNING'` (T_STRING), expecting `','` or `')'` on line number 5
+
+ - syntax error, unexpected `'"Divide(): The value '0' as a '` (T_CONSTANT_ENCAPSED_STRING) on line number 5
+
+
+/// type=MS, answer=[4,5]
+
+Which statements correctly describe the error?
+
+ - There is no semicolon `;` at the end of the statement on line 5.
+
+ - There is no open parenthesis `(` after `trigger_error` on line 5.
+
+ - There is no open curly brace `{` after the `if` construct on line 4.
+
+ - On line 5, the function call `trigger_error( , E_USER_WARNING)` is invalid.
+
+ - There is no error message specified as the first argument of the `trigger_error()` function call on line 5.
+
+:::
+
+
+/// type=CR, answer=[tests/ErrorHandling/MissingCommaBetweenArgumentsTest.php]
+
+Correct the code so that it outputs the string `Warning: Divide(): The value '0' as a divisor is not allowed in /index.php on line 5`.
+
+```php
+<?php
+function Divide($dividend, $divisor)
+{
+    if ($divisor == 0) {
+        trigger_error(, E_USER_WARNING);
+        return false;
+    } else {
+        return($dividend / $divisor);
+    }
+}
+
+echo Divide(10, 0);
+?>
+```
+
+
+:::
+
+/// type=REPL, readonly=true
+
+```php
+<?php
+function Divide($dividend, $divisor)
+{
+    return($dividend / $divisor);
+}
+
+function myErrorHandler($error_no, $error_msg)
+{
+    echo "Error: [$error_no] " . $error_msg; 
+
+    die("\nPlease try again."); 
+}
+
+set_error_handler(myErrorHandler);
+echo Divide(10, 0);
+?>
+```
+/// type=SS, answer=[4]
+
+Execute the program. What is the error message?
+
+ - syntax error, unexpected `','` on line number 5
+
+ - Warning: Divide(): The value '0' as a divisor is not allowed in /index.php on line 5
+
+ - syntax error, unexpected `'$error_file'` (T_VARIABLE), expecting `')'` on line number 12
+
+ - Notice: Use of undefined constant `myErrorHandler` - assumed `'myErrorHandler'` on line number 14
```

Comment:
This was displayed on my end:

![image](https://user-images.githubusercontent.com/57437877/75420276-a950a580-5972-11ea-87d8-feaadbdf43bf.png)

Is the `Error: [2] Division by zero` an error message as well?

Reply:
Actually this `Error: [2] Division by zero` is not the error message but this is generated from the `echo` statement provided in our `myErrorHandler()` function. Please check. Thank you.